services/BasicTypes.proto

```diff
@@ -378,9 +378,11 @@ message NodeAddressBook {
 }
 
 message SemanticVersion {
-   int32 major = 1; // Increases with incompatible API changes
-   int32 minor = 2; // Increases with backwards-compatible new functionality
-   int32 patch = 3; // Increases with backwards-compatible bug fixes
+    int32 major = 1; // Increases with incompatible API changes
+    int32 minor = 2; // Increases with backwards-compatible new functionality
+    int32 patch = 3; // Increases with backwards-compatible bug fixes
+    string pre_release_version = 4; // A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers (https://semver.org/#spec-item-9); so given a semver 0.14.0-alpha.1+21AF26D3, this field would contain 'alpha.1'
```

Comment:
Appending `_version` is redundant as this is within the `SemanticVersion` context. Also none of the others have that appended. Would prefer just `pre_release`.

Reply:
The [semver spec](https://semver.org/#spec-item-9) refers to it as a "pre-release version", and that's within the context of semver also. :D

Reply:
I recommend following something like this: https://docs.rs/semver/0.11.0/semver/struct.Version.html#fields

 - `pre`
 - `build`

To keep things concise. It's obvious what those are, we don't need the `_release_version` suffix.

Reply:
> The [semver spec](https://semver.org/#spec-item-9) refers to it as a "pre-release version", and that's within the context of semver also. 😁

In the paragraph before they also say `Major version` yet we don't have that field named as `major_version`. That's inconsistent. As that is a paragraph in English it is okay to add the version context. For a field name, it looks weird to have version twice in `semanticVersion.getPreReleaseVersion()`.

Reply:
Cool, shortened field names to `pre` and `build`!